packages/design-system/src/components/tooltip/index.js

```diff
@@ -133,6 +134,17 @@ function Tooltip({
 
     setArrowDelta(delta);
   }, []);
+  // When near the bottom of the viewport and the tooltip is placed on the bottom we want to force the tooltip to the top as to not
+  // cutoff the contents of the tooltip.
+  const positionPlacement = useCallback(() => {
+    const tooltipElBoundingBox = tooltipRef.current?.getBoundingClientRect();
```

Comment:
Since this needs to be in a callback in order to get the size of the tooltip, I am assuming I needed the [`useEffectLayout` in `Popup`](https://github.com/google/web-stories-wp/pull/8945/files#diff-ceecde430ba66db8a84fe3b7d8d7be935229b867156e208d0589e29364ee6d0aR152)

Reply:
So I just learned this (thanks to Will), but I believe you can actually pass a callback ref to get this data:
https://reactjs.org/docs/refs-and-the-dom.html#callback-refs
https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780

I'm sure this works fine as is tho too

Reply:
Oh yes, I remember seeing this! That must be why I felt a little iffy on my implementation. 🙇🏻‍♀️